controllers/configmap_controller.go

```diff
@@ -51,12 +53,30 @@ type ConfigMapReconciler struct {
 //+kubebuilder:rbac:groups=metallb.io,resources=bfdprofiles,verbs=get;list;watch;create;
 //+kubebuilder:rbac:groups=metallb.io,resources=bfdprofiles/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=metallb.io,resources=addresspools/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=metallb.io,resources=metallbs,verbs=get;list;watch;
 
 func (r *ConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	r.Log.Info(fmt.Sprintf("Starting ConfigMap reconcile loop for %v", req.NamespacedName))
 	defer r.Log.Info(fmt.Sprintf("Finish ConfigMap reconcile loop for %v", req.NamespacedName))
 
-	err := reconcileConfigMap(ctx, r.Client, r.Log, r.Namespace)
+	instance := &metallbv1beta1.MetalLB{}
```

Comment:
If possible, I think a better way do approach this is to set an owner reference between the metallb crd and the configmap (like we do in https://github.com/openshift/metallb-operator/blob/26adecd8f161110c82b12e8e3fd49e8b0433cb4a/controllers/metallb_controller.go#L180-L181). 
This will make everything works automatically, because the k8s gc will take care of that (as we do with the deployment, for example).

https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/

Reply:
@fedepaol Yes, I've tried this approach and somehow owner reference (attempted both SetControllerReference and SetOwnerReference) is not set, Does the owner ref really work for config maps ? 
Anyway even if we make it work, ConfigMapReconciler kicks upon config map delete and creates it again. 
Hence I tried this change in `configmap_controller.go` and it works, but it mandates metallb crd object to be present for creating `config` cm object (which makes sense, i guess) and when metallb crd object is removed then `config` cm object is also removed. Of course it needs update in unit tests and e2e tests.

Reply:
@fedepaol The owner reference actually works here, the issue was with `reconcileConfigMap`  logic which resets config map without an owner reference. now fixed it, does this look ok ? I'll go update unit and e2e tests based on the feedback.